### ISSUE TITLE
Wrong CRC calculation in upgrade_crc_check

### ISSUE THREAD

**Issue by hammer-dp-ua** [espressif/ESP8266_IOT_PLATFORM#9]
Please rollback a change made in 10.25.2016 08:27:15 by SeregaKai.
abs function was placed not just by mistake. Looks like it converts "unsigned int" to "int" inside itself and then really changes bitwise representation if int result is negative.

**Comment by borro0** [comment]
I agree on this. The gen_appbin.py used to generate the crc in the first place, also uses abs.
![image](https://user-images.githubusercontent.com/15669405/43061952-7e2e03f8-8e57-11e8-8a1c-c8c5a14920eb.png)
